File: C++/Medium/Accepted/find-all-groups-of-farmland_8.cpp
class Solution {
public:
    vector<vector<int>> findFarmland(vector<vector<int>>& land) {
        
        vector<vector<int>> result;
        int m = land.size(), n = land[0].size();
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (land[i][j] == 1) {
                    int right = j;
                    while (right < n && land[i][right] == 1) {
                        right++;
                    }
                    int down = i;
                    while (down < m && land[down][j] == 1) {
                        down++;
                    }
                    result.push_back({i, j, down - 1, right - 1});
                    for (int k = i; k < down; k++) {
                        for (int l = j; l < right; l++) {
                            land[k][l] = 0;
                        }
                    }
                }
            }
        }
        return result;
    }
};

File: Java/Medium/WrongAnswer/find-all-groups-of-farmland_8.java
class Solution {
    public int[][] findFarmland(int[][] land) {
        
        int m = land.length;
        int n = land[0].length;
        boolean[][] visited = new boolean[m][n];
        List<int[]> res = new ArrayList<>();
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                if(visited[i][j] || land[i][j] == 0) continue;
                int[] temp = new int[4];
                temp[0] = i;
                temp[1] = j;
                int x = i;
                int y = j;
                while(x < m && land[x][y] == 1){
                    x++;
                }
                temp[2] = x - 1;
                while(y < n && land[x - 1][y] == 1){
                    y++;
                }
                temp[3] = y - 1;
                res.add(temp);
            }
        }
        int[][] ans = new int[res.size()][4];
        for(int i = 0; i < res.size(); i++){
            ans[i] = res.get(i);
        }
        return ans;
    }
}